app/models/task.py

```diff
@@ -1,5 +1,40 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import ForeignKey
+from typing import Optional
 from ..db import db
+from datetime import datetime
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+  from .goal import Goal
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[datetime] = mapped_column(default=None, nullable=True)
```

Comment:
If we want a column to be `nullable` then we would want to use `Optional` like you did on `line 15`. Without `Optional` we don't need `nullable=False`.